### ISSUE TITLE
Enabling *sticky* or *layer shift for next action* causes key to do nothing

### ISSUE THREAD

**Issue by heralden** [keyboardio/Chrysalis#819]
**Describe the bug**
When you set a key to be a modifier or shift to layer, it works as expected. As soon as you enable sticky for the modifier, or set it to *layer shift for next action*, it will acquire the behaviour of a blank key. Pressing the key followed by a character key does not cause the modifier or layer shift to apply, nor does holding it while pressing a character key.

**To Reproduce**
1. Set a key to be a modifier (I only tested with Shift)
2. Enable Sticky in the Modifiers accordion
3. Click Save Changes
4. Try using the modifier - both its oneshot/sticky and hold behaviour

The above can also be done with *layer shift for next action* instead of a modifier.

**Expected behavior**
The key should apply its modifier or layer shift to the next key pressed. If held, the modifier or layer shift should apply to keys pressed while held. If tapped twice, the modifier or layer shift should continue to apply until tapped again (can't remember if this is default behaviour in Chrysalis).

**Desktop (please complete the following information):**
 - OS: macOS Monterey 12.3
 - Chrysalis Version: 0.9.4-snapshot.184, 0.9.3 and 0.9.2 tested

**Additional context**
Not sure if this bug was introduced in one of the new 0.9.* versions, if it only occurs when run on Apple Silicon or if it's specific to the new Model 100 keyboard.

I attached the chrysalis debug bundle JSON (I had to rename it to .txt to upload it here).

[chrysalis-debug.bundle.txt](https://github.com/keyboardio/Chrysalis/files/8765441/chrysalis-debug.bundle.txt)

**Comment by algernon** [comment]
This is caused by the firmware not having the necessary plugins enabled. Fortunately, with recent firmware, we can ask the firmware which plugins it is compiled with, and we can pop up a warning when we encounter keys that use features unavailable in the current firmware.

I would not disable setting such keys up, because setting up a keymap for a future firmware is an entirely valid thing. But we could pop up a notification bar.

On top of that, the next version of Chrysalis will ship with firmware that has these plugins (Qukeys & OneShot) enabled either in the default, or the experimental firmware.

**Comment by heralden** [comment]
I totally forgot about the experimental firmware! I used the Model100/experimental.bin file from here: https://github.com/keyboardio/Chrysalis-Firmware-Bundle/releases

When I use it as the custom firmware for the firmware update in Chrysalis I get this error.

<img width="328" alt="Screenshot 2022-05-24 at 20 54 05" src="https://user-images.githubusercontent.com/3865590/170122231-ea7ce6d8-229b-4879-9b12-da19932d4506.png">

**Comment by algernon** [comment]
Yep, that is a known issue, Chrysalis is a bit wonky when it comes to flashing Model100. Hang on tight, we'll have a fix soon.

**Comment by heralden** [comment]
As it was a JS error, I thought I'd try with 0.9.2 instead of 0.9.3 and this time it worked, although I still got the `Error flashing the firmware` message after it was frozen for 30 seconds (the keys never started flashing red). After replugging the keyboard, I had to restore my keymap and when I tested OSM it worked. Not sure if using a different version made a difference or I just got lucky this time.

Thanks for pointing me in the right direction!
